### ISSUE TITLE
wav2vec 2.0 pretraining training stopping condition?

### ISSUE THREAD

**Issue by machakos23** [facebookresearch/fairseq#3979]
Hi, I know base trains towards 400k updates. I am currently at 250k and I have noticed the following:
- training and validation **accuracy** is still going up 
-  training and validation **loss** is still decreasing
- **prob_perplexity** and **code_perplexity** have peaked and have been decreasing since then 

Question is, should I continue training despite the perplexities decreasing? They were steadily increasing since the start of pretraining but have since started reversing. Is there is any risk of overtraining considering I am still seeing regular progress on every epoch? (i.e an improved checkpoint_best, decreasing losses and improving accuracy)

Some extra background:
- Training on 2000h of audio 
- Used an older pretrained model as a base (but with more training data this time) so it's not training completely from scratch. 
- Have been manipulating `lr` and `update_freq` through training. Initially started with the default for `lr` and 64 for the latter but that would have taken too long, so after 9k updates, I dropped it down to 1 for the first 90k updates with a smaller `lr`. Since then it's been at 8 with the same `lr` as before for 1.
